xbmc/dbwrappers/DatabaseQuery.cpp

```diff
@@ -16,43 +16,50 @@
 #include "utils/Variant.h"
 #include "utils/XBMCTinyXML.h"
 
-typedef struct
+#include <algorithm>
+#include <array>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace
 {
-  char string[15];
-  CDatabaseQueryRule::SEARCH_OPERATOR op;
-  int localizedString;
-} operatorField;
-
-static const operatorField operators[] = {
-    {"contains", CDatabaseQueryRule::OPERATOR_CONTAINS, 21400},
-    {"doesnotcontain", CDatabaseQueryRule::OPERATOR_DOES_NOT_CONTAIN, 21401},
-    {"is", CDatabaseQueryRule::OPERATOR_EQUALS, 21402},
-    {"isnot", CDatabaseQueryRule::OPERATOR_DOES_NOT_EQUAL, 21403},
-    {"startswith", CDatabaseQueryRule::OPERATOR_STARTS_WITH, 21404},
-    {"endswith", CDatabaseQueryRule::OPERATOR_ENDS_WITH, 21405},
-    {"greaterthan", CDatabaseQueryRule::OPERATOR_GREATER_THAN, 21406},
-    {"lessthan", CDatabaseQueryRule::OPERATOR_LESS_THAN, 21407},
-    {"after", CDatabaseQueryRule::OPERATOR_AFTER, 21408},
-    {"before", CDatabaseQueryRule::OPERATOR_BEFORE, 21409},
-    {"inthelast", CDatabaseQueryRule::OPERATOR_IN_THE_LAST, 21410},
-    {"notinthelast", CDatabaseQueryRule::OPERATOR_NOT_IN_THE_LAST, 21411},
-    {"true", CDatabaseQueryRule::OPERATOR_TRUE, 20122},
-    {"false", CDatabaseQueryRule::OPERATOR_FALSE, 20424},
-    {"between", CDatabaseQueryRule::OPERATOR_BETWEEN, 21456}};
-
-CDatabaseQueryRule::CDatabaseQueryRule()
+struct OperatorField
 {
-  m_field = 0;
-  m_operator = OPERATOR_CONTAINS;
-}
+  std::string name;
+  CDatabaseQueryRule::SEARCH_OPERATOR op{CDatabaseQueryRule::OPERATOR_START};
+  int localizedName{0};
+};
+
+// clang-format off
+const std::array<OperatorField, 15> operators = {{
```

Comment:
Could be made constexpr with a string_view name?

Reply:
`string_view name`needs more adjustments to compile, which I wanted to avoid. The SQ PRs are already big enough. It's always the question where to draw a line.

I will give `constexpr` a go, but IIRC correctly I tried already and it did not compile on all platforms.

Reply:
I saw but that's strange.